src/sentry/integrations/slack/message_builder/notifications.py

```diff
@@ -1,109 +1,62 @@
         self.context = context
         self.recipient = recipient
 
+    def build(self) -> SlackBody:
+        return self._build(
+            title=self.notification.build_attachment_title(),
+            title_link=self.notification.get_title_link(),
+            text=self.notification.get_message_description(),
+            footer=self.notification.build_notification_footer(self.recipient),
+            actions=self.notification.get_message_actions(),
+        )
+
 
-class SlackProjectNotificationsMessageBuilder(SlackNotificationsMessageBuilder):
+class IssueNotificationMessageBuilder(SlackNotificationsMessageBuilder):
     def __init__(
         self,
         notification: ProjectNotification,
         context: Mapping[str, Any],
-        recipient: Union["Team", "User"],
+        recipient: Team | User,
     ) -> None:
         super().__init__(notification, context, recipient)
-        # TODO: use generics here to do this
         self.notification: ProjectNotification = notification
 
     def build(self) -> SlackBody:
         group = getattr(self.notification, "group", None)
-        # TODO: refactor so we don't call SlackIssuesMessageBuilder through SlackProjectNotificationsMessageBuilder
-        if self.notification.is_message_issue_unfurl:
-            return SlackIssuesMessageBuilder(
-                group=group,
-                event=getattr(self.notification, "event", None),
-                tags=self.context.get("tags", None),
-                rules=getattr(self.notification, "rules", None),
-                issue_details=True,
-                notification=self.notification,
-                recipient=self.recipient,
-            ).build()
-
-        if isinstance(self.notification, ReleaseActivityNotification):
-            return self._build(
-                text="",
-                actions=build_deploy_buttons(self.notification),
-                footer=build_notification_footer(self.notification, self.recipient),
-            )
-
-        if isinstance(self.notification, NewProcessingIssuesActivityNotification):
-            return self._build(
-                title=self.notification.get_title(),
-                text=self.notification.get_message_description(),
-                footer=build_notification_footer(self.notification, self.recipient),
-            )
-
-        return self._build(
-            title=build_attachment_title(group),
-            title_link=get_title_link(group, None, False, True, self.notification),
-            text=self.notification.get_message_description(),
-            footer=build_notification_footer(self.notification, self.recipient),
-            color="info",
-        )
-
-
-def build_notification_attachment(
-    notification: BaseNotification,
-    context: Mapping[str, Any],
-    recipient: Union["Team", "User"],
-) -> SlackBody:
-    """@deprecated"""
-    return notification.build_slack_attachment(context, recipient)
+        return SlackIssuesMessageBuilder(
```

Comment:
@mgaeta Is this going to be a temporary solution or is this a pattern you plan on sticking with? Cause I don't think having message builders call each other is a great long-term solution.